File: examples/vulnerable_test.go
// Go test file with intentional vulnerabilities

package main

import (
    "database/sql"
    "fmt"
    "os/exec"
    "net/http"
    "crypto/md5"
)

// CWE-78: Command Injection
func executeCommand(userInput string) {
    cmd := exec.Command("sh", "-c", "ls -la "+userInput)
    cmd.Run()
}

// CWE-89: SQL Injection
func queryUser(db *sql.DB, username string) {
    query := "SELECT * FROM users WHERE name = '" + username + "'"
    db.Query(query)
}

// CWE-327: Weak Crypto
func hashPassword(password string) string {
    hasher := md5.New()
    hasher.Write([]byte(password))
    return string(hasher.Sum(nil))
}

// CWE-798: Hardcoded Credentials
func connectDB() {
    password := "MyPassword123"
    apiKey := "sk-1234567890abcdef"
    // Use credentials
    fmt.Println(password, apiKey)
}

// CWE-79: XSS
func handleRequest(w http.ResponseWriter, r *http.Request) {
    name := r.URL.Query().Get("name")
    fmt.Fprintf(w, "<h1>Hello, %s</h1>", name)
}

File: examples/vulnerable_test.java
// Java test file with intentional vulnerabilities for testing

import java.io.*;
import java.sql.*;
import javax.servlet.http.*;

public class VulnerableApp {
    
    // CWE-78: Command Injection
    public void executeCommand(String userInput) {
        try {
            Runtime.getRuntime().exec("ls -la " + userInput);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
    
    // CWE-89: SQL Injection
    public void queryDatabase(String username) {
        try {
            Connection conn = DriverManager.getConnection("jdbc:mysql://localhost/db");
            Statement stmt = conn.createStatement();
            String query = "SELECT * FROM users WHERE username = '" + username + "'";
            ResultSet rs = stmt.executeQuery(query);
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
    
    // CWE-22: Path Traversal
    public void readFile(HttpServletRequest request) {
        String filename = request.getParameter("file");
        try {
            File file = new File("/var/www/files/" + filename);
            BufferedReader reader = new BufferedReader(new FileReader(file));
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
    
    // CWE-327: Weak Cryptography
    public String hashPassword(String password) {
        try {
            MessageDigest md = MessageDigest.getInstance("MD5");
            byte[] hash = md.digest(password.getBytes());
            return new String(hash);
        } catch (Exception e) {
            return null;
        }
    }
    
    // CWE-798: Hardcoded Credentials
    public void connectToDatabase() {
        String password = "SuperSecret123!";
        String apiKey = "AIzaSyD1234567890abcdefghijk";
        // Use credentials...
    }
}


